### PR TITLE
feat(gpu): add NVIDIA and AMD device plugin daemonsets

### PR DESCRIPTION
**Pull request by rushtehrani**
Resolves https://github.com/onepanelio/core/issues/98

### CODE REVIEW

File: gpu-plugins/overlays/amd/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../base
+- daemonset-amdgpu-device-plugin-daemonset-kube-system.yaml
```
Comment (Vafilor) [inline gpu-plugins/overlays/amd/kustomization.yaml]: We have daemonset in the file name twice, maybe remove one of them unless there's a formatting reason?

Comment (rushtehrani) [reply gpu-plugins/overlays/amd/kustomization.yaml]: I know it's a bit weird but it's following the convention we decided on: `<resource-kind>-<resource-name>-<namespace>.yaml`. In this case, `resource-name` has `daemonset` in it.

Comment (Vafilor) [reply gpu-plugins/overlays/amd/kustomization.yaml]: Gotcha. Cool, as long as there's a reason for it.

Comment (aleksandrmelnikov) [reply gpu-plugins/overlays/amd/kustomization.yaml]: To reduce the number of "-", I've been combining the pieces for my naming.
Example: "daemonset-amdgpudeviceplugindaemonset-kubesystem.yaml"

**Pull request merged**